### PR TITLE
Adds a delete local file action to FileListEditor menu

### PR DESCRIPTION
**Pull request by stefan-kolb**
In case the auto downloader saved a wrong file, the user might need to delete it #101.

### CODE REVIEW

**Comment by simonharrer** [comment]
Hm, one idea: should there be a warning if the file could not be removed? This can happen when the file is still open in adobe acrobat. As a user I would like to close adobe acrobat and retry the deletion again.

**Comment by stefan-kolb** [comment]
Good point. But I can only catch `IOException` for this which is rather generic?!

**Comment by simonharrer** [comment]
Hm, maybe check after deletion if the file is still there?

**Comment by simonharrer** [comment]
I would suggest:

1. Try to delete file
2. Only if the file has been deleted successfully, unlink

This algorithm also includes the case that the file is not there before, reducing the operation to simply unlink. 

This ensures that the user can retry and simulates a transaction. In case of an error, you can simply show a message through a popup.

File: src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java [hunk 1]
```diff
@@ -238,6 +242,34 @@ public void actionPerformed(ActionEvent e) {
...
+                FileListEntry entry = tableModel.getEntry(row);
+                Path filePath = FileUtil.expandFilename(metaData, entry.getLink()).toPath();
+
+                try {
+                    // transactional delete and unlink
+                    if(Files.exists(filePath)) {
+                        Files.delete(filePath);
+                        removeEntries();
+                    } else {
+                        removeEntries();
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java]: move `removeEntries` out of the condition as will always be executed.

**Comment by simonharrer** [comment]
:+1:

**Pull request merged**